Tests.WebView/CommonTests.cs

```diff
@@ -0,0 +1,194 @@
+                var taskCompletionSourceListener21 = new TaskCompletionSource<bool>();
+                var taskCompletionSourceListener22 = new TaskCompletionSource<bool>();
+
+                var listener1 = TargetView.AttachListener("event1_name");
+                listener1.Handler += delegate {
+                    listener1Counter++;
+                    taskCompletionSourceListener1.SetResult(true);
+                };
+
+                var listener2 = TargetView.AttachListener("event2_name");
+                listener2.Handler += delegate {
+                    listener2Counter++;
+                    taskCompletionSourceListener21.SetResult(true);
+                };
+                listener2.Handler += delegate {
+                    listener2Counter++;
+                    taskCompletionSourceListener22.SetResult(true);
+                };
+
+                await Load($"<html><script>{listener1}{listener2}</script><body></body></html>");
+                Task.WaitAll(taskCompletionSourceListener1.Task, taskCompletionSourceListener21.Task, taskCompletionSourceListener22.Task);
+
+                Assert.AreEqual(1, listener1Counter);
+                Assert.AreEqual(2, listener2Counter);
+            });
+        }
+
+        [Test(Description = "Attached listeners are called in Dispatcher thread")]
+        public async Task ListenersAreCalledInDispatcherThread() {
+            await Run(async () => {
+                var taskCompletionSource = new TaskCompletionSource<bool>();
+                bool? canAccessDispatcher = null;
+
+                var listener = TargetView.AttachListener("event_name");
+                listener.UIHandler += delegate {
+                    canAccessDispatcher = Dispatcher.UIThread.CheckAccess();
+                    taskCompletionSource.SetResult(true);
+                };
+
+                await Load($"<html><script>{listener}</script><body></body></html>");
+                await taskCompletionSource.Task;
+
+                Assert.IsTrue(canAccessDispatcher);
+            });
+        }
+
+        [Test(Description = "Unhandled Exception event is called when an async unhandled error occurs inside a listener")]
+        public async Task UnhandledExceptionEventIsCalledOnListenerError() {
+            await Run(() => {
+                var taskCompletionSource = new TaskCompletionSource<bool>();
+                const string ExceptionMessage = "hey";
+                Exception exception = null;
+
+                WithUnhandledExceptionHandling(async () => {
+                    var listener = TargetView.AttachListener("event_name");
+                    listener.Handler += delegate {
+                        taskCompletionSource.SetResult(true);
+                        throw new Exception(ExceptionMessage);
+                    };
+
+                    await Load($"<html><script>{listener}</script><body></body></html>");
+                    await taskCompletionSource.Task;
```

Comment:
I think this test will be flaky because you're setting the task result before the exception is thrown .. it should be set inside the exception handler